src/pymor/algorithms/phdmd.py

```diff
@@ -0,0 +1,564 @@
+    obtained from `X` itself.
+
+    The formal description of this algorithm can be found in :cite:`MMU22`.
+
+    Parameters
+    ----------
+    X
+        The |VectorArray| of the system state data.
+    Y
+        The |VectorArray| of the system output data.
+    U
+        The |VectorArray| of the system control data.
+    Xdot
+        If not `None`, the |VectorArray| of the system time derivative data.
+        If `None`, `dt` will be used to estimate this data from `X` using an implicit midpoint rule.
+    dt
+        See `Xdot`.
+    H
+        If not `None`, the `NumPy array` representing the systems Hamiltonian.
+    initial_J
+        If not `None`, the `Numpy array` representing the initial guess for the :math:`J` operator.
+    initial_R
+        If not `None`, the `Numpy array` representing the initial guess for the :math:`R` operator.
+    initial_alpha
+        The initial value used in calculating the update step.
+    miniter
+        Minimum amount of iterations to perform.
+    maxiter
+        Fail if the iteration count reaches this value without converging.
+    atol
+        Finish when the absolute error measure is below this threshold.
+    rtol
+        Finish when the relative update error measure is below this threshold.
+    weighted_rtol
+        In the weighted pH DMD initialization problem, truncate when the relative value is
+        below this threshold.
+    skew_procrustes_rtol
+        In the skew-symmetric Procrustes problem, truncate when the relative value is
+        below this threshold.
+
+    Returns
+    -------
+    inf_model
+        The inferred |PHLTIModel|.
+    data
+        Dict containing the following fields:
+
+            :update_norms:          |NumPy array| of the norms of the update vectors after
+                                    each iteration.
+            :abs_errs:              |NumPy array| of the absolute errors after each iteration.
+            :rel_errs:              |NumPy array| of the relative  errors after each iteration.
+            :iterations:            Number of total pH DMD iterations.
+            :skew_procrustes_data:  `list` of data generated by the skew-symmetric
+                                    Procrustes subproblems.
+            :weighted_init_data:    If `initial_J` and `initial_R` are both `None`, dict of data
+                                    generated by the weighted pH DMD initialization subproblem.
+            :skew_init_data:        If only `initial_J` is `None`, dict of data generated by the
+                                    skew-symmetric Procrustes subproblem.
+            :spsd_init_data:        If only `initial_R` is `None`, dict of data generated by the psd
+                                    symmetric Procrustes subproblem.
+    """
+    assert isinstance(X, VectorArray)
+    assert isinstance(Y, VectorArray)
+    assert isinstance(U, VectorArray)
+    assert len(X) == len(Y)
+    assert len(X) % len(U) == len(Y) % len(U) == 0
```

Comment:
```suggestion
    assert len(X) % len(U) == 0
```

It is already asserted that `len(X)` and `len(Y)` are equal.